gcloud-java-pubsub/src/test/java/com/google/cloud/pubsub/ReceivedMessageTest.java

```diff
@@ -0,0 +1,188 @@
+/*
+ * Copyright 2016 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.pubsub;
+
+import static org.easymock.EasyMock.createMock;
+import static org.easymock.EasyMock.createStrictMock;
+import static org.easymock.EasyMock.expect;
+import static org.easymock.EasyMock.replay;
+import static org.easymock.EasyMock.verify;
+import static org.junit.Assert.assertArrayEquals;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.assertSame;
+
+import com.google.api.client.util.Charsets;
+import com.google.cloud.ByteArray;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.util.concurrent.Futures;
+
+import org.easymock.EasyMock;
+import org.junit.After;
+import org.junit.Test;
+
+import java.nio.charset.StandardCharsets;
+import java.util.Map;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+
+public class ReceivedMessageTest {
+
+  private static final String SUBSCRIPTION = "subscription";
+  private static final String ACK_ID = "ackId";
+  private static final String MESSAGE_ID = "messageId";
+  private static final String PAYLOAD_STRING = "payload";
+  private static final ByteArray PAYLOAD =
+      ByteArray.copyFrom("payload".getBytes(StandardCharsets.UTF_8));
+  private static final Map<String, String> ATTRIBUTES =
+      ImmutableMap.of("key1", "value1", "key2", "value2");
+  private static final Long PUBLISH_TIME = 42L;
+  private static final Message MESSAGE = Message.builder(PAYLOAD)
+      .id(MESSAGE_ID)
+      .attributes(ATTRIBUTES)
+      .publishTime(PUBLISH_TIME)
+      .build();
+  private static final com.google.pubsub.v1.ReceivedMessage RECEIVED_MESSAGE_PB =
+      com.google.pubsub.v1.ReceivedMessage.newBuilder()
+          .setMessage(MESSAGE.toPb())
+          .setAckId(ACK_ID)
+          .build();
+
+  private PubSub serviceMockReturnsOptions = createStrictMock(PubSub.class);
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Private field 'serviceMockReturnsOptions' could be made final; it is only initialized in the declaration or constructor.](https://www.codacy.com/app/mziccard/gcloud-java/file/2671233162/issues/source?bid=3123109&fileBranchId=3332594#l65)